sweep/sweeper.go

```diff
@@ -1399,12 +1401,22 @@ func (s *UtxoSweeper) handleNewInput(input *sweepInputMessage) {
 	pi.ntfnRegCancel = cancel
 }
 
-// attachAvailableRBFInfo queries the mempool to see whether the given input
-// has already been spent. If so, it will query the sweeper store to fetch the
-// fee info of the spending transction, hence preparing for possible RBF.
-func (s *UtxoSweeper) attachAvailableRBFInfo(pi *pendingInput) *pendingInput {
+// decideStateAndRBFInfo queries the mempool to see whether the given input has
+// already been spent. If so, the state Published will be returned, otherwise
+// state Init. When spent, it will query the sweeper store to fetch the fee
+// info of the spending transction, and construct an RBFInfo based on it.
+// Suppose an error occurs, fn.None is returned.
+func (s *UtxoSweeper) decideStateAndRBFInfo(op wire.OutPoint) (
+	SweepState, fn.Option[RBFInfo]) {
+
 	// Check if we can find the spending tx of this input in mempool.
-	txOption := s.mempoolLookup(*pi.OutPoint())
+	txOption := s.mempoolLookup(op)
+
+	// Extract the spending tx from the option.
+	var tx *wire.MsgTx
+	txOption.WhenSome(func(t wire.MsgTx) {
```

Comment:
Can alternatively write this as:
```go
tx := txOption.UnwrapOr(nil) 
```

Reply:
switched!

Reply:
hmmm seems we need to update `fn` mod, gonna defer this change in later PRs.